### DISCUSSION TITLE
N64 CRC error

### DISCUSSION THREAD

**Discussion by Ruggerio150** [sanni/cartreader#143] [Need help]
![A8B4C6BE-EC04-4E3B-B79B-3979BF5B3071](https://user-images.githubusercontent.com/82999194/116784615-7a155700-aa95-11eb-8ecc-bf9eacbb5b6c.jpeg)
Since the 6.1 Firmware i get an crc error of Reading PAL games. I try 2 difference Games. 
did someone know what to do?
Thanks

**Comment by sanni** [comment]
Try quick formatting the SD card using [SD Memory Card Formatter](https://www.sdcard.org/downloads/formatter/) with CHS format size adjustment enabled.

**Reply by Ruggerio150** [reply]
I use an Sandisc 32GB and CHS ist not available.

**Reply by Ruggerio150** [reply]
The Error is just at N64. The other Systems worked without CRC Error. I also trying an SD card Adapter with an 4GB SD card. Same error as with the microsd Adapter and 32GB Microsd Card

**Reply by Ruggerio150** [reply]
I Test with Firmware 5.5 and other SD Card. Same error. I am out of ideas

**Comment by sanni** [comment]
Is it just the two games or every N64 game? Have you checked the dumps on your PC to see if maybe they are correct but just a different version than in the database?

Other than that make sure the N64 game's gold fingers are cleaned with a q-tip and IPA and that you insert the game perfectly centered onto the N64 slot.

**Reply by Ruggerio150** [reply]
Many thanks for the answer. It's actually in every N64 game. Once he did not show any errors when reading. I read out the module again, then there was another CRC error. the CRC code also changes more often when it is read out again. I have already cleaned the contacts with isopropanol. What is the N64 slot sleeve for? With him I don't get the games attached, it's too tight then. How can I test the dumps on the PC?

**Comment by sanni** [comment]
It's very easy to misaligned N64 carts when plugging them into the Cart Reader. The N64 sleeve prevents this by removing the gaps left and right between the slot and the cart. But since the Chinese manufacturers sell different N64 slots the sleeve might not fit always. Then you have to manually center the cartridge or you can wrap the slot with some electric tape to make it thicker so the N64 can't be misaligned.

If you still get dump errors after checking that the cartridge is perfectly centered then we need to check the hardware.

- Please check the voltage between those two points, it needs to be 3.3V when the switch is in 3V mode, you can use a multimeter to do so:

![image](https://dl.dropboxusercontent.com/s/b583u9dlkhoiwk8/voltageeee.jpg?dl=1)

- Also make sure you followed the [build guide](https://github.com/sanni/cartreader/wiki/How-to-build-the-PCB) and did not forget to desolder the 5V regulator and the transistor from the Arduino board:
![image](https://dl.dropboxusercontent.com/s/38ad01xbej3k1rx/0.png?dl=1)

**Reply by Ruggerio150** [reply]
Thank you again,  i will test it when I am at home. The Problem with the sleeve is, it doesn’t fit into the cartridge. On the slot it fit. I will send an picture later with the sleeve on the cartridge

**Reply by sanni** [reply]
Then most likely something went wrong when 3D printing. The sleeve should be printed in "vase mode" and only be one line thick (~0.4mm).

**Reply by Ruggerio150** [reply]
I have now re-printed the n64 sleeve and it now fits perfectly. I also measured the supply. 3.28 volts. but unfortunately still CRC errors 😭

**Comment by sanni** [comment]
Well all that is left to do now is to download the [pinout.xls](https://github.com/sanni/cartreader/blob/master/pinout.xls?raw=true) and measure the resistance between each pin on the cartridge slot(column N64 rom) and the corresponding pin on the Arduino Mega(column Arduino). The resistance of each connection should be below 1 ohm. If one pin is higher, re-solder that connection.
